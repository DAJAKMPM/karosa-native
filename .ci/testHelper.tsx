import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
