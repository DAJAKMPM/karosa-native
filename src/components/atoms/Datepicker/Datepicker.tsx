/**
 *
 * Datepicker
 * @format
 *
 */

import React, { FC } from "react";
import { TextInputMask } from "react-native-masked-text";

import type { PropsType } from "./types";
import { DATE_FORMAT } from "./config";

const Datepicker: FC<PropsType> = (props) => {
  const { value, onChange, placeholderColor, style } = props;

  return (
    <TextInputMask
      style={style}
      placeholder={DATE_FORMAT}
      type="datetime"
      options={{
        format: DATE_FORMAT,
      }}
      placeholderTextColor={placeholderColor}
      value={value}
      returnKeyType="done"
      onChangeText={onChange}
    />
  );
};

export default Datepicker;
