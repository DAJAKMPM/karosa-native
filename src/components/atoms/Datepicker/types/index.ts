/**
 *
 * @format
 *
 */

import { StyleProp, TextStyle } from "react-native";
import { ChangeText } from "@app/utils";

export type PropsType = {
  value: string;
  onChange: ChangeText;
  placeholderColor?: string;
  style?: StyleProp<TextStyle>;
};
