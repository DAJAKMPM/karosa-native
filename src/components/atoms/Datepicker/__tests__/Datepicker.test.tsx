import React from "react";
import { noop } from "lodash";
import { renderWithWrapper } from "@app/ci/testHelper";
import Datepicker from "../Datepicker";

describe("Datepicker Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Datepicker value="2022/06/07" onChange={(_text) => noop()} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
