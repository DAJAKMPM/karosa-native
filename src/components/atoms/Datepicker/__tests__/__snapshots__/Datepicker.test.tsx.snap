// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    options={
      Object {
        "format": "YYYY/MM/DD",
      }
    }
    placeholder="YYYY/MM/DD"
    rejectResponderTermination={true}
    returnKeyType="done"
    type="datetime"
    underlineColorAndroid="transparent"
    value="2022/06/07"
  />
</View>
`;
